File: handlers/signup.py
#!/usr/bin/env Python
# coding=utf-8

import tornado.web
from methods.db import *

class SignupHandler(tornado.web.RequestHandler):
    def get(self):
        self.render("signup.html")

    def post(self):
        zusername = self.get_argument("zusername")
        zpassword = self.get_argument("zpassword")
        zname = self.get_argument("zname")
        zmailbox = self.get_argument("zmailbox")
        zoccupation = self.get_argument("zoccupation")
        zcompany = self.get_argument("zcompany")
        sql = "select * from users where username ='" + zusername + "'"
        cur.execute(sql)
        lines = cur.fetchall()
        if lines:
            self.write("error")
        else:
            try:
                sql="INSERT INTO users(username,password,name,mailbox,occupation,company) VALUES (" + zusername +","+ zpassword +","+ zname +","+ zmailbox +","+ zoccupation +","+ zcompany +")"
                cur.execute(sql)
                conn.commit()
            except:
                conn.rollback()
                return false   
            self.write("ok")      
        
        
        

File: static/js/signup.js
$(document).ready(function(){
    $("#signup").click(function(){
        var username=$("#zusername").val();
        var pwd=$("#zpassword1").val();
        var name=$("#zname").val();
        var mail=$("#zmailbox").val();
        var occ=$("#zoccupation").val();
        var company=$("#zcompany").val();
        var pd = {"zusername":username,"zpassword":pwd,"zname":name,"zmailbox":mail,"zoccupation":occ,"zcompany":company};
        if($("#zpassword1").val()==$("#zpassword2").val()){
            $.ajax({
                type:"post",
                url:"/signup",
                data:pd,
                cache:false,
                success:function(data){
                    if(data=="ok"){
                        alert("注册成功！");
                        window.location.href = "/";
                    }
                    else if(data=="error"){
                        alert("该用户名已经存在！");
                    }
                },
                error:function(){
                    alert("error!");
                },
            });
        }
        else{
            alert("两次输入的密码不一致，请重新输入");
        }
    });
});